Add Inline Java UDFs (#2605)
confluentinc/ksql: docs/developer-guide/syntax-reference.rst

The WITH clause supports the following properties:

+-------------------------+--------------------------------------------------------------------------------------------+
| Property                | Description                                                                                |
+=========================+============================================================================================+
| AUTHOR                  | The author of the function                                                                 |
+-------------------------+--------------------------------------------------------------------------------------------+
| DESCRIPTION             | A description of the function                                                              |
+-------------------------+--------------------------------------------------------------------------------------------+
| VERSION                 | The version (e.g. 0.1.0) of the function                                                   |
+-------------------------+--------------------------------------------------------------------------------------------+

> Example of createing a simple function. **Note:** the arguments are passed as an array named **args**

Reviewer: ```suggestion
Example of creating a simple function. **Note:** the arguments are passed as an array named *args*
```

> Example of createing a simple function. **Note:** the arguments are passed as an array named **args**

Reviewer: ```suggestion
Example of creating a simple function. The arguments are passed as an array named *args*
```

to the inline script.

.. code:: sql

    CREATE OR REPLACE FUNCTION GREET(name STRING)
    RETURNS STRING
    LANGUAGE JAVA
    AS BEGIN
        return "Hello " + args[0]; \
    END;

> Here is a slightly more complicated inline UDF that imports some classes from the standard library and

Reviewer: ```suggestion
Here's a more complex inline UDF that imports some classes from the standard library and
```

    :start-after: Avro_note_start
    :end-before: Avro_note_end

.. create-function:

CREATE [OR REPLACE] FUNCTION
----------------------------

**Synopsis**

.. code:: sql

    CREATE (OR REPLACE?) FUNCTION function_name ( { field_name data_type } [, ...] )
    RETURNS data_type
    LANGUAGE language_name
    WITH ( property_name = expression [, ...] );
    AS $$
        inline_script
    $$;

**Description**

Create a new inline UDF with the specified arguments and properties.

> Note: currently, the only supported language is **JAVA**.

Reviewer: ```suggestion
Currently, the only supported language is Java.
```

If the OR REPLACE clause is present, the statement doesn't fail if the function

> already exists. Instead, the existing function will be replaced. Note: if an

Reviewer: ```suggestion
already exists. Instead, the existing function replaced. If an
```

> active query is using a UDF that was replaced, it will continue to use the

Reviewer: ```suggestion
active query is using a UDF that was replaced, it continues to use the
```

The WITH clause supports the following properties:

+-------------------------+--------------------------------------------------------------------------------------------+
| Property                | Description                                                                                |
+=========================+============================================================================================+
| AUTHOR                  | The author of the function                                                                 |
+-------------------------+--------------------------------------------------------------------------------------------+
| DESCRIPTION             | A description of the function                                                              |
+-------------------------+--------------------------------------------------------------------------------------------+
| VERSION                 | The version (e.g. 0.1.0) of the function                                                   |
+-------------------------+--------------------------------------------------------------------------------------------+

> The following example creates a simple function. Note: the arguments are

Reviewer: ```suggestion
The following example creates a simple function. The arguments are
```

converted to uppercase when passed to the inline script.

> Also, when defining an inline UDF in the KSQL CLI, you will need to escape

Reviewer: ```suggestion
Also, when defining an inline UDF in the KSQL CLI, you must escape
```

If the IF EXISTS clause is present, the statement doesn't fail if the table
doesn't exist.

DROP FUNCTION [IF EXISTS];
--------------------------

**Synopsis**

.. code:: sql

    DROP FUNCTION [IF EXISTS] function_name;

**Description**

> Drop an inline UDF from the function registry. Note: you cannot drop UDFs that were not

Reviewer: ```suggestion
Drop an inline UDF from the function registry. You can't drop UDFs that weren't
```

> created via the CREATE [OR REPLACE] FUNCTION query. Furthermore, if an active query

Reviewer: ```suggestion
created by using the CREATE [OR REPLACE] FUNCTION query. Also, if an active query
```

> is using a UDF that was dropped, it will continue to use the underlying UDF instance

Reviewer: ```suggestion
is using a UDF that was dropped, it continues to use the underlying UDF instance
```
